docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+
+Upon startup, objects from ui, parser and storage are created. Prompted for user input, Duke receives the "editp"
+command which is forwarded to the parser to be interpreted respectively. Once the `EditPatientCommand` object is 
+created, it retrieves the patient index to edit the existing patient information from the patientList. 
+The `Patient` class is called by its setter method, `setPatientInfo()`, to update the fields as provided by the user. 
+This ensures that the encapsulated variables such as age, name, contact number and address are not only enforced but
+also protected. 
+
+##### 2.2.1.2 Design Considerations
+###### Aspect: Data Type for Appointment's Date and Time
+
++ Alternative 1 (current choice): Store as a String
+    * Pros: 
+        - Easier implementation
+        - Greater user flexibility
+    * Cons:
+        -  Cannot sort if needed
+
++ Alternative 2: Store as Date
+    * Pros:
+        - Has flexibility to parse or format date using existing methods available for use.
+    * Cons:
+        - Immutable-value classes mean it is not thread-safe (using Java.util.SimpleDateFormat).
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.2 Converter Module
+The converter module consists of one class which converts the format of date and time using a custom format defined by
+special formatting characters (ie. SimpleDateFormat). This class is primarily used to format a user-input date and time
+in the `Appointment` class. As illustrated below, its methods are called during the creation of the `Appointment`
+object constructor. 
+
+![](images/SD_Converter.png)
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.3 BRANDON storage module
+
+The Storage module consists of 3 different classes. 
+The PatientList and AppointmentList classes act as data structures to store the records of Patient and Appointment 
+objects respectively. They function as ADTs, where various commands from Command objects can manipulate the records within.
+The Storage class manages the load and save operations involving the PatientList and PatientList class. 
+These operations are usually invoked on startup, whenever changes are made to the ADTs and before exiting the program.
+The class diagram for the storage module is as seen below: 
+
+
+![](images/storageclass.PNG)
+
+&nbsp;
+
+On startup, Duke invokes the loadSavedAppointment() and loadSavedPatient() methods in Storage. This allows the program 
+to retrieve previously stored data from a .txt file and convert it into the static AppointmentList and PatientList objects for use
+within the program. 
+
+The Storage object creates a Scanner object that will parse individual lines in the .txt file, convert them into
+new Appointments, and then add them to an ArrayList called `appointmentListToReturn`. This `appointmentListToReturn` will be passed back to Duke to
+construct the static AppointmentList. The sequence diagram is shown below:
+
+![](images/loadsavedappt_seq.PNG)
```

Comment:
![image](https://user-images.githubusercontent.com/31772358/77993413-ae7e8700-735a-11ea-8266-43e44691b2af.png)

Maybe `appointmentListToReturn` and `s` should have activation bars for constructors?